org.eclipse.xtend.core/src/org/eclipse/xtend/core/compiler/XtendGenerator.xtend

```diff
@@ -320,7 +330,9 @@ class XtendGenerator extends JvmModelGenerator implements IGenerator2 {
 			}
 			if (declaringType.local && it instanceof JvmOperation) {
 				val declarator = declaringType as JvmGenericType
-				if (!declarator.anonymous) {
+				if (!declarator.anonymous ||
```

Comment:
I'm not sure whether I like the fact that the helper doesn't check the declarator itself but this code here has to do both.

Reply:
Indeed I had put a Javadoc comment on the helper https://github.com/eclipse/xtext/pull/2898/files/bc766b0068465148679855a36c8c9c88e5c62065#diff-c6ea2a981e27e70dc2cd53ff982ee051ac65096607807462e10196d625518f88R35
to avoid a check there, but if you prefer we can check `isAnonymous` directly in the helper, if that's what you mean.

Reply:
@szarnekow I've just pushed https://github.com/eclipse/xtext/pull/2898/commits/6af742c8bceb96792ea6d06325536ed8b0a31209 

that check for anonymous is redundant